internalsdk/pro/http.go

```diff
@@ -1,25 +0,0 @@
-package pro
-
-import (
-	"net/http"
-	"time"
-
-	"github.com/getlantern/flashlight/v7/proxied"
-)
-
-// NewHTTPClient creates a new http.Client that is configured to use the given options and http.RoundTripper wrapped with
-// proxied.AsRoundTripper to process requests
-func NewHTTPClient(rt http.RoundTripper, timeout time.Duration) *http.Client {
-	if timeout == 0 {
-		timeout = 30 * time.Second
-	}
-	return &http.Client{
-		Transport: proxied.AsRoundTripper(
-			func(req *http.Request) (*http.Response, error) {
-				log.Tracef("Pro client processing request to: %v (%v)", req.Host, req.URL.Host)
-				return rt.RoundTrip(req)
-			},
-		),
```

Comment:
This doesn't really do anything.  `proxied.AsRoundTripper` is a utility function which simply converts a function to a `http.RoundTripper`.  The request is not necessarily proxied.  IMO `proxied.AsRoundTripper` is misleading and should never have been added to the public API of the `proxied` package.

I removed `NewHTTPClient` because it only adds complexity and reader overhead.  Instead of calls like:
```
client := NewHTTPClient(someTransport, timeout)
```
It's more straight-forward and readable to say:
```
client := &http.Client{
    Transport: someRoundTripper,
    Timeout: timeout,
}
```
I understand that this is opinionated.  Let me know if you'd like me to revert this change.

Reply:
If we shouldn't be using proxied.AsRoundTripper, is it possible to remove it from the proxied package on the flashlight PR and any remaining references to it from the internalsdk? In particular, we need to update replica [here](https://github.com/getlantern/lantern-client/blob/main/internalsdk/replica.go#L153).